asserts/snapasserts/validation_sets.go

```diff
@@ -189,6 +189,22 @@ type ValidationSets struct {
 	snaps map[string]*snapContraints
 }
 
+// Revisions returns the set of snap revisions that is enforced by the
+// validation sets that ValidationSets manages. ValidationSets.Conflict should
+// be checked before calling this method.
+func (v *ValidationSets) Revisions() map[string]snap.Revision {
+	snapNameToRevision := make(map[string]snap.Revision, len(v.snaps))
+	for _, sn := range v.snaps {
+		for revision := range sn.revisions {
+			if revision == invalidPresRevision {
+				continue
+			}
+			snapNameToRevision[sn.name] = revision
```

Comment:
What would happen if we have more than one revision here? How would we choose the right one if we have multiple validation assertions with the same snap with different revisions? Is that a valid case?